### ISSUE TITLE
Failure to generate cert from CSR generated by FreeIPA

### ISSUE THREAD

**Issue by james-masson** [cloudflare/cfssl#346]
I'm trying to sign a CSR generated from FreeIPA - http://www.freeipa.org/page/FreeIPAv2:Certificate_Authority#Use_a_Different_CA_to_sign_the_IPA_CA_certificate

The CSR that is generated is always rejected by `cfssl` - no matter what domain/hostname is used in the CSR generation process.

```
> cat jcbe-ipa.foo.example.com.csr | cfssl sign -
{"code":9003,"message":"x509: extension attribute contained non-OCTET STRING data"}

> cfssl sign -csr="jcbe-ipa.foo.example.com.csr"
{"code":9003,"message":"x509: extension attribute contained non-OCTET STRING data"}
```

This seems to be a Golang SSL parsing error. https://golang.org/src/crypto/x509/x509.go

CSR below.
```
-----BEGIN CERTIFICATE REQUEST-----
MIICsTCCAZkCAQAwOjEYMBYGA1UECgwPRk9PLkVYQU1QTEUuQ09NMR4wHAYDVQQDDBVDZXJ0aWZp
Y2F0ZSBBdXRob3JpdHkwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDP6+PlbKReGO5G
2x5WU6TEgPtVb5CZsXdt2Df7rUmbz9SMxhE+u6DdU0qI1q02hzQ2WszAL3nxm7+QazLWNAHpXyan
b/w7ESDZ0m8hivaK1Nd5nm0WWXQ8/y1HpdXeDV0hXQtpn1dLlKtVd/G1KTlqCSuNhRF+DmBx4efD
a2DXgNzRmECJII9LlIkeOwH4EO8f985UdLjurOZt6A4J4SW3IPAXU+h3ZG2oguM+Pd2+3oB0hO4B
Q1+D06awS4cdmMbJxDST1FtNzeIJvsu1aLDQwvVR74cZVzOLEfASIoQzhn+6Zk8ORvLaxP0RMT/z
aG5eJkOHTY5dP05icFU11STHAgMBAAGgMjAwBgkqhkiG9w0BCQ4xIzAhMA8GA1UdEwEB/wQFMAMB
Af8wDgYDVR0PAQH/BAQDAgHGMA0GCSqGSIb3DQEBCwUAA4IBAQBjrF9bIh3qds41c0RrBdy64Bze
8QeN0kDzQMg2WX3T1VDR3SZ/so/6X0O2kZeaWxpHNewxdsMEgnlSIHsExz6UO5+M999l6GQmQPJN
unJyysNw7UQxt9DPwwkHTFvvZoeod9ZQE0ayauLZdhmlqoHIE2pUu4eg3h9l6fjjO6HQLTlV2t+6
WEW0WCXKxVFlB0MvCQOEbTlMQAWdo1AwktBBQYjQ66slUz2h56VmaW+b2BGKlOzdwuyKlrFDTN4D
NR2AQEtAuG17u3bVy5Lu6nF4zzw+zCLBMB58hk6VGit7kkTaFRL57hYv2KwQ3IRyj2WRj7B78FGa
2jRgGCvrPlzG
-----END CERTIFICATE REQUEST-----
```

Any ideas what's going wrong? Is the CSR malformed? OpenSSL seems quite happy with it.

**Comment by kisom** [comment]
For some reason, the extension value isn't being parsed. I'll look into it; it looks like an ASN.1 issue.

**Comment by jacobhaven** [comment]
This is solved in Go 1.6 by https://github.com/golang/go/commit/e78e654c1de0a7bfe0314d6954d42b046f14f1bb

**Comment by kisom** [comment]
Closing because this is an issue with Go, and it appears to be fixed in Go 1.6.
